.github/actions/run-ios-simulator/src/main.ts

```diff
@@ -0,0 +1,74 @@
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+import { v4 as uuidv4 } from "uuid";
+import * as childProcess from "promisify-child-process";
+
+async function run(): Promise<void> {
+    const id = uuidv4().split("-").join("");
+
+    try {
+        const appPath = core.getInput("appPath", { required: true });
+        const bundleId = core.getInput("bundleId", { required: true });
+        const iphoneToSimulate = core.getInput("iphoneToSimulate", { required: false });
+        const args = core.getInput("arguments", { required: false });
+
+        let runtimeId = "";
+        const options : exec.ExecOptions = {};
+        options.listeners = {
+            stdout: (data: Buffer) => {
+                runtimeId += data.toString();
+            },
+        };
+
+        await execCmd("xcrun simctl list runtimes", options);
+
+        // Sample output: iOS 14.5 (14.5 - 18E182) - com.apple.CoreSimulator.SimRuntime.iOS-14-5
+        // and we want to extract "iOS 14.5"
+        // If we want to allow launching watchOS/tvOS simulators, replace the 'iOS' with an 'os' argument
+        const matches = runtimeId.match(/(iOS \d{1,2}(.\d{1,2})?)/g);
+        if (matches && matches.length > 0) {
+            runtimeId = matches[0].replace(" ", "");
+            core.info(`runtimeId: ${runtimeId}`);
+        }
+
+        try {
+            await execCmd(`xcrun simctl create ${id} com.apple.CoreSimulator.SimDeviceType.${iphoneToSimulate} ${runtimeId.toString()}`);
+        }
+        catch {
+            // Different combinantions of xcode and macOS versions have shown different syntax acceptance about the runtime, therefore 1 last attempt with a different syntax.
+            // additionally, this uses  childProcess.exec instead of @actions/exec because the latter doesn't properly parse the pipe (|) char
+            const { stdout, stderr } = await childProcess.exec("xcrun simctl list runtimes |  awk '/com.apple.CoreSimulator.SimRuntime.iOS/ { match($0, /com.apple.CoreSimulator.SimRuntime.iOS-[0-9.-]+/); print substr($0, RSTART, RLENGTH); exit }'");
```

Comment:
Should this use `execCmd`?

Reply:
Also, we already ran `xcrun simctl list runtimes` - rather than running it a second time and matching with `awk`, we can write a regex that parses both formats for the runtime and try to create a simulator with either one.

Reply:
It doesn't use execCmd because as explained in the comment, the pipe `|` isn't properly parsed by what's at the core of execCmd. The fact that I didn't rework the awk command is simply because I wasn't sure about the usefulness of this "second try" and I wanted to see the opinion of the reviewer. If you think it makes sense to retry with a different runtime syntax, I'll re-write the line using just a regex.

Reply:
If I understand the `awk` syntax correctly, it's just a regex match against the runtime string. We can combine both matches (this one and the one in line 28) into a single check:

```js
const match =/(?<runtime1>iOS \d{1,2}(.\d{1,2})?).*(?<runtime2>com\.apple\.CoreSimulator\.SimRuntime\.iOS-[0-9.-]+)/g.exec(runtimeId);

try {
   await execCmd(`xcrun simctl create ${id} com.apple.CoreSimulator.SimDeviceType.${iphoneToSimulate} ${match.groups.runtime1}`);
}
catch {
   await execCmd(`xcrun simctl create ${id} com.apple.CoreSimulator.SimDeviceType.${iphoneToSimulate} ${match.groups.runtime2}`);
}
```